sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationChecker.scala

```diff
@@ -41,41 +42,70 @@ object UnsupportedOperationChecker extends Logging {
     }
   }
 
+  private def hasRangeExprAgainstEventTimeCol(e: Expression): Boolean = e.exists {
+    case neq @ (_: LessThanOrEqual | _: LessThan | _: GreaterThanOrEqual | _: GreaterThan) =>
+      hasEventTimeColNeq(neq)
+    case _ => false
+  }
+
+  private def hasEventTimeColNeq(neq: Expression): Boolean = {
```

Comment:
What does 'Neq' stand for here? - 'not equal'? How does this method know that that is the case - seems like it can be applied to any binary expression. Maybe add an assert?

Reply:
This function is only used in `hasRangeExprAgainstEventTimeCol`. Yes it does mean not equal, maybe rename it to `hasEventTimeColBinaryComp` makes more sense. 

I did try to change function signature to be `private def hasEventTimeColNeq(neq: BinaryComparison): Boolean` but the compiler would complain because here
```
case neq @ (_: LessThanOrEqual | _: LessThan | _: GreaterThanOrEqual | _: GreaterThan)
```
neq can only be identified as Expression.

I'll just put these two helper functions under `hasRangeExprAgainstEventTimeCol`, that makes the logic more clear.